src/modules/mc_hover_thrust_estimator/MulticopterHoverThrustEstimator.cpp

```diff
@@ -59,7 +59,7 @@ MulticopterHoverThrustEstimator::~MulticopterHoverThrustEstimator()
 
 bool MulticopterHoverThrustEstimator::init()
 {
-	if (!_vehicle_local_position_setpoint_sub.registerCallback()) {
+	if (!_vehicle_local_position_sub.registerCallback()) {
```

Comment:
There's a (potential) small scheduling issue to be aware of now. 

Previously by running off of vehicle_local_position_setpoint publications this was guaranteed to run right after mc_pos_control, consuming the same vehicle_local_position message that was just used by the controller to compute the new thrust.

It's now possible that they run out of sync with mc_hover_thrust_estimator running before mc_pos_control using the new acceleration (vehicle_local_position) with the previous thrust (vehicle_local_position_setpoint). Is this potential worst case tolerable for now or worth digging into?

For various reasons I was going to invent some kind of WorkItem priority mechanism (or start order), but I haven't gotten around to it yet.

Reply:
Thanks, noted. However, this isn't a strict requirement as there is a delay+lag between the thrust setpoint and the actual generated thrust/acceleration that we are currently not taking in account anyway. The filter is robust enough to deal with that and the small scheduling issue won't cause a problem.